src/compiler/checker.ts

```diff
@@ -37271,14 +37492,25 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return false;
     }
 
-    function mergeInferences(target: InferenceInfo[], source: InferenceInfo[]) {
+    function mergeInferences(target: InferenceInfo[], source: InferenceInfo[], clone = false) {

```

Comment:
For consistency with `mergeInferenceContexts` I'd call this one `cloneInferences`
```suggestion
    function mergeInferences(target: InferenceInfo[], source: InferenceInfo[], cloneInferences = false) {
```